### PR TITLE
adopt viper to manage configuration

### PR DESCRIPTION
**Pull request by nickhsine**
@babygoat

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -1,31 +1,62 @@
...
+		"port":     "3306",
+	})
+
+	viper.SetDefault("encryptsettings.salt", "salt")
 
-	// Load config file
-	err := utils.LoadConfig(fname)
 	if err != nil {
-		log.Fatal("main.load_config.fatal_error: ", err.Error())
+		panic(fmt.Errorf("Fatal error config file: %s \n", err))
```
Comment (babygoat) [inline main.go]: I suggest the error handling should be move up to below Line#25

**Comment by babygoat** [comment]
After discussion:
* config.json -> config.yaml
* Use viper to control all the configs without defining all the configuration data structure (tradeoff: performance downgrade)

**Comment by nickhsine** [comment]
After thinking, I define those structs and use global variable to store the config.

**Comment by nickhsine** [comment]
@babygoat please help to review the code.

**Comment by nickhsine** [comment]
Fixed according to  review.

**Comment by babygoat** [comment]
LGTM

**Pull request merged**